### DISCUSSION TITLE
How to deal with image hash

### DISCUSSION THREAD

**Discussion by linonetwo** [vzhd1701/evernote-backup#53] [Q&A] [answered]
In .enex , image will become

```
<en-media style="height: auto;" hash="698008cc6d9fd53005d5c2459037dd8c" type="image/jpeg"></en-media>
```

and

```
<resource><data encoding="base64">
JVBERi0xLjQgU2hhcnAgU2Nhbm5lZCBJbWFnZVBERgolU2hhcnAgTm9uLUVuY3J5cHRpb24KMyAwIG9i
ago8PAovVHlwZSAvUGFnZQovUGFyZW50IDEgMCBSCi9SZXNvdXJjZXMgNCAwIFIKL0NvbnRlbnRzIDUg
MCBSCi9NZWRpYUJveCBbMCAwIDU5NSA4NDFdCj4+CmVuZG9iago0IDAgb2JqCjw8Ci9Qcm9jU2V0IFsv
UERGIC9JbWFnZUNdCi9YT2JqZWN0IDw8IC9JbWcxIDYgMCBSID4+Cj4+CmVuZG9iago1IDAgb2JqCjw8
...
```

And the `<resource>` part don't have a hash. Manually decode base64 + md5 it gives wrong result

![图片](https://github.com/vzhd1701/evernote-backup/assets/3746270/fdffb782-9b20-45ed-928e-ec0b8305a7cc)

https://github.com/emn178/js-md5/issues/36#issuecomment-1762886759

So it is impossible to make it markdown to tiddlywiki or obsidian now.

**Comment by vzhd1701** [comment]
Could you please provide a full example?

**Reply by linonetwo** [reply]
[网络文库.enex.zip](https://github.com/vzhd1701/evernote-backup/files/13024924/enex.zip)

Try this with many images.

**Reply by linonetwo** [reply]
The problem is not having hash. Anyone want to use it can manually apply patch in https://github.com/vzhd1701/evernote-backup/pull/54 to your local python script.

**Reply by vzhd1701** [reply]
I checked all images and resources in your enex file, they are all matching OK. Here is a test script:
https://gist.github.com/vzhd1701/20a81fc6294382b428a26a01c6cd6cda

**Reply by linonetwo** [reply]
Sorry, I know hash are matching the resource, I was mean it is inconvenient to match resource and hash in frontend JS (don't have build-in md5 method, and some of implementation is buggy https://github.com/emn178/js-md5/issues/36

**Comment by linonetwo** [comment, marked as answer]
Use this script provided by @vzhd1701 https://github.com/vzhd1701/evernote-backup/pull/54#issuecomment-1768569911

**Reply by linonetwo** [reply]
I will point Evernote migration guide to this script.
